### ISSUE TITLE
About csi_trace

### ISSUE THREAD

**Issue by haimapi** [dhalperi/linux-80211n-csitool-supplementary#120]
Dear all,
        Yesterday, when I processed CSI data with matlab on another laptop, a question occurred:
![clipboard](https://cloud.githubusercontent.com/assets/15070162/11229469/72d8dbc0-8dd0-11e5-8fda-468db11bf53f.png)

I've read FAQ and recompile read_bfee.c with the command:

> > > mex read_bfee.c

![clipboard](https://cloud.githubusercontent.com/assets/15070162/11229495/b9d654c6-8dd0-11e5-8c09-0a4a32786bc4.png)

It reported errors.... So, where did I ignore and how to resolve this? THX!

**Comment by ozawali** [comment]
you should recomplie see FAQ 2A

**Comment by haimapi** [comment]
@ozawali , I just want to know how to recompile readf_bfee.c.... Isn't it "mex read_bfee.c"?

**Comment by ozawali** [comment]
right !

**Comment by haimapi** [comment]
@ozawali , yeah, I did as told, after typing "mex read_bfee.c" it threw out errors and recompile failed... 

Did I forget something? Everything is normal on my former laptop...

**Comment by dhalperi** [comment]
What version of MATLAB are you using?

I've only compiled the file using a specific, fairly new, MATLAB on OS X. There may be platform-dependent changes you need to make to the source `read_bfee.c` file.

**Comment by dpward** [comment]
@haimapi @dhalperi The error is caused because your C compiler (presumably Microsoft Visual C++) does not support non-constant initializers for arrays, which was [introduced in C99](http://www.drdobbs.com/the-new-c-declarations-initializations/184401377).

Microsoft Visual C++ has long been lacking in support for C99 (at least [until very recently](http://blogs.msdn.com/b/vcblog/archive/2013/07/19/c99-library-support-in-visual-studio-2013.aspx)). That being said, I'm actually not sure why I didn't see this error before when I last built the MEX-file for Windows (is this an older version of MATLAB / Windows / Microsoft Visual C++?).

I will see if I can modify that code soon so it is C89 compliant and will work for you (my apologies that I've been very inactive on this lately).

**Comment by haimapi** [comment]
@dhalperi @dpward thx very much for your help! the matlab version i use is '2010b'

**Comment by asifhanif1122** [comment]
I have noticed that code does not work on Matlab versions except for 2010 version. If anyone is experiencing this "??? Undefined command/function 'read_bfee' " error, he/she should use Matlab 2010 version.

**Comment by egaebel** [comment]
So it isn't possible to compile on any version except 2010a??? There must be some workaround...

**Comment by asifhanif1122** [comment]
I experienced these problems while executing files on 64 bit version of Matlab. It is possible that files may work on other versions of Matlab (32-Bit).

**Comment by egaebel** [comment]
Update: I was able to successfully compile on Matlab 2014a, 64-bit. So I don't think this is a general problem

**Comment by zainalabidenakhtar** [comment]
i am also facing the same problem but not got any solution yet. if any body got this problem and then resolved he may kindly guide

**Comment by zainalabidenakhtar** [comment]
> > csi_trace = read_bf_file('sample_data/log.all_csi.6.7.6');
> > ??? Undefined function or method 'read_bfee' for input arguments of type 'uint8'.

Error in ==> read_bf_file at 66
        ret{count} = read_bfee(bytes);

> > mex read_bfee.c

Select a compiler: 

[0] None 

Compiler: read_bfee
Please select from 0-0 
Compiler:

**Comment by zainalabidenakhtar** [comment]
@haimapi i am using 64-bit MATLAB 7.8.0.347 (R2009a),  (64-bit Window-10) and facing the same issue if u find any solution please guide me

**Comment by haimapi** [comment]
@zainalabidenakhtar seems that what it matter is the version of matlab, try 32-bit matlab :-)

**Comment by zainalabidenakhtar** [comment]
@haimapi my problem was resolved after using 32-bit windows with 32-bit MATLAB :). i think some extra files are required for 64-bit MATLAB

**Comment by PigSea** [comment]
@haimapi @zainalabidenakhtar 
I use matlab2012a (64bit) and visual studio 2010,since the tool only provide the .mexw32 version of read_bfee, when i recompile readbfee.c, i also encount some problems. I slove it by moving all the definitions of variables to the front of the file and get the .mexw64 version sucessfully. Hope it's useful.

**Comment by tfsanam** [comment]
@egaebel Can you please tell me what compiler did you use for Matlab R2014a 64bit? Also have you changed anything on the C source code read_bfee.c file?

**Comment by egaebel** [comment]
I'm not sure what you mean by what compiler did I use for Matlab R2014a 64 bit. I used that version of matlab to compile and run my code. If I recall correctly I made some minor changes to read_bfee.c but I do not recall specifically.
